### ISSUE TITLE
Daemon crashes due to a merkle mask issue

### ISSUE THREAD

**Issue by o1pranay** [CodaProtocol/coda#2969]
```
  ☠  Coda Daemon crashed. The Coda Protocol developers would like to know why!

  Please:
    Open an issue:
      https://github.com/CodaProtocol/coda/issues/new

    Briefly describe what you were doing, and include the last 20 lines from .coda-config/coda.log.
    And then paste the following:

    (monitor.ml.Error
  (Failure
    "Couldn't unregister mask with UUID 0e8d45b1-fe70-305e-4375-ab4b1e74dd74 from parent e6f51a1d-9c2f-3815-a744-30ab374f49ba, mask not registered with that parent")
  ("Raised at file \"stdlib.ml\", line 33, characters 17-33"
    "Called from file \"src/lib/merkle_mask/maskable_merkle_tree.ml\", line 159, characters 12-72"
    "Called from file \"src/lib/coda_base/ledger.ml\" (inlined), line 167, characters 35-79"
    "Called from file \"src/lib/transition_frontier/transition_frontier.ml\", line 784, characters 24-63"
    "Called from file \"src/lib/transition_frontier/transition_frontier.ml\", line 787, characters 14-35"
    "Called from file \"list.ml\", line 106, characters 12-15"
    "Called from file \"src/list0.ml\" (inlined), line 26, characters 40-75"
    "Called from file \"src/lib/transition_frontier/transition_frontier.ml\", line 803, characters 12-66"
    "Called from file \"src/lib/transition_handler/processor.ml\", line 64, characters 17-51"
    "Called from file \"src/lib/transition_handler/processor.ml\", line 149, characters 8-129"
    "Called from file \"src/deferred0.ml\", line 56, characters 64-69"
    "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5"
    "Called from file \"src/job_queue.ml\", line 170, characters 6-47"
    "Caught by monitor coda"))
```

coda.log: https://gist.github.com/o1pranay/2579499710a710b1c6d9179c6bae2df2
registered_masks.dot: https://gist.github.com/o1pranay/b938a5a1dec7c1a852ef7741800ad14d
frontier.dot: https://gist.github.com/o1pranay/b47baf6b0c1f83cfe9217ffcff721cad

**Comment by o1pranay** [comment]
Happened again - was preceded by this

```
2019-07-24 05:18:18 UTC [Info] Timed out waiting for the parent of $cached_transition after 600000 ms, signalling a catchup job
	cached_transition: {
  "protocol_state": {
    "previous_state_hash":
      "30043940010010806182625808329235426367057326882686941686404102658465933051175015297961967918506642700036494951636340653507725997734284024631124122940978966849008977573681625165820114928431227397925352112657035980616218238195523",
    "body": {
      "blockchain_state": {
        "staged_ledger_hash": {
          "non_snark": {
            "ledger_hash":
              "719668189824584934993574463800560343633911623009983453171502994648138900542852971474133427309936903356627789006285908055076022694031540707582156833720242229524444438584075148861690566662880572309021088256538469335722737105567",
            "aux_hash":
              "8SQNSzeGBNJcdXHLGAbXtxCTvX6nP9qW7wbjQBNtKgiaMsKDXredz",
            "pending_coinbase_aux":
              "5Lt1pKMfAPWAaqbW3mvNUy5w1ZbHaYtdNNEjLUAvVnAYDfqKZ2M"
          },
          "pending_coinbase_hash":
            "32489900214151365326872642890202978200201898370956315489798390358731159497501179434757453984797354745761718719683787600756891866291857937559522289508857155313787484809741164043523424178657946446810360550738440380732446759047745"
        },
        "snarked_ledger_hash":
          "38932695040467596106866402734597443669063381446628167838929129222958650359402660018169492542114228579857016809663362878702967691140114883880635297843337526424344049903879831696969707777320618038873222528382381697025453708056747",
        "timestamp": "1563944700143"
      },
      "consensus_state": {
        "blockchain_length": "23",
        "epoch_count": "0",
        "min_epoch_length": "72",
        "last_vrf_output":
          "8aXHoBq1SXFGg29Wx1wmX1BKT3DQFc5owvDzSjEeyP2Ua3rwfX",
        "total_currency": 10109980,
        "curr_epoch": "0",
        "curr_slot": "31",
        "staking_epoch_data": {
          "ledger": {
            "hash":
              "38932695040467596106866402734597443669063381446628167838929129222958650359402660018169492542114228579857016809663362878702967691140114883880635297843337526424344049903879831696969707777320618038873222528382381697025453708056747",
            "total_currency": 10109980
          },
          "seed":
            "2071893303198007985737678972190309212568452221625132024511988170095494148670997278812694070338313361389889122280160253462982652030041813566301365289695187505618174204273471887226695702458395861269694368663558765191107385382142",
          "start_checkpoint":
            "2071893303198007985737678972190309212568452221625132024511988170095494148670997278812694070338313361389889122280160253462982652030041813566301365289695187505618174204273471887226695702458395861269694368663558765191107385382142",
          "lock_checkpoint":
            "2071893303198007985737678972190309212568452221625132024511988170095494148670997278812694070338313361389889122280160253462982652030041813566301365289695187505618174204273471887226695702458395861269694368663558765191107385382142",
          "epoch_length": "1"
        },
        "next_epoch_data": {
          "ledger": {
            "hash":
              "38932695040467596106866402734597443669063381446628167838929129222958650359402660018169492542114228579857016809663362878702967691140114883880635297843337526424344049903879831696969707777320618038873222528382381697025453708056747",
            "total_currency": 10109980
          },
          "seed":
            "24289248328964365496268591521656653856602237050780355335366540327609565535206100634016174628978537511933703312861421647101135519983037587237558266368514578605538146560750505496769581366718360228581254149632293927003335192527860",
          "start_checkpoint":
            "2071893303198007985737678972190309212568452221625132024511988170095494148670997278812694070338313361389889122280160253462982652030041813566301365289695187505618174204273471887226695702458395861269694368663558765191107385382142",
          "lock_checkpoint":
            "30043940010010806182625808329235426367057326882686941686404102658465933051175015297961967918506642700036494951636340653507725997734284024631124122940978966849008977573681625165820114928431227397925352112657035980616218238195523",
          "epoch_length": "24"
        },
        "has_ancestor_in_same_checkpoint_window": true,
        "checkpoints": {
          "data": {
            "prefix": [
              "36782869935965414222803766111589106059928653266006562368237491770920419223196351921290099024431045678696270371899077719754582744324683342090263286949885333652891185465016898405769609259698781267121722885844488370666255009246289"
            ],
            "tail": "0"
          },
          "hash":
            "12062717558525538699047780862059193266924556459560851006160860343216576123787521205271672413556467062262639340546367611173236998289502441061812779352640732323492364704217327960088370937391626808596131208187008001457662385788416"
        }
      }
    }
  },
  "protocol_state_proof": "<opaque>",
  "staged_ledger_diff": "<opaque>"
}
2019-07-24 05:18:22 UTC [Warn] Not rebroadcasting block $state_hash because it was received "4 slots too late"
	state_hash: "16823378300780972795670848723637315391307885870363770767609128949910057919173214098563571342333384761041886261255655308109950987588546795571988459400561515443962741443133190465781021765649970652966511797409840893419367624298689"
2019-07-24 05:18:22 UTC [Warn] Not rebroadcasting block $state_hash because it was received "3 slots too late"
	state_hash: "30043940010010806182625808329235426367057326882686941686404102658465933051175015297961967918506642700036494951636340653507725997734284024631124122940978966849008977573681625165820114928431227397925352112657035980616218238195523"
2019-07-24 05:18:22 UTC [Warn] Not rebroadcasting block $state_hash because it was received "2 slots too late"
	state_hash: "22444349808069136341513286754383612158790135108157189535026105356900136636913275989379524964871787786094512958126929010840111419659505988410039856028143989414755513068578948739832360625916534704456222493135469903040295976660845"
2019-07-24 05:18:22 UTC [Warn] Not rebroadcasting block $state_hash because it was received "2 slots too late"
	state_hash: "3233100825834559220516201868164877049321133515134676994520305813190987554437781378961419925954187862960913793689495942145285435011331571754762661382490525460383493718620603956073842386765477981750898625410132338176335790964508"
2019-07-24 05:18:22 UTC [Warn] Not rebroadcasting block $state_hash because it was received "2 slots too late"
	state_hash: "20433419504081982912877182428001638231879636064755780203466818994519726899130521959467088070402440807778843065700261620247599142268055893766568146080035590550379596718173471728744699410931348991191500501904586778253654757797954"
2019-07-24 05:19:06 UTC [Info] Connected to some peers [[host : 98.210.109.223, discovery_port : 8303, communication_port : 8302]]
2019-07-24 05:22:11 UTC [Info] Some peers disconnected [host : 98.210.109.223, discovery_port : 8303, communication_port : 8302]
2019-07-24 05:22:11 UTC [Info] Removing peer from peer set: [host : 98.210.109.223, discovery_port : 8303, communication_port : 8302]
2019-07-24 05:22:16 UTC [Info] Connected to some peers [[host : 98.210.109.223, discovery_port : 8303, communication_port : 8302]]


  ☠  Coda Daemon crashed. The Coda Protocol developers would like to know why!

  Please:
    Open an issue:
      https://github.com/CodaProtocol/coda/issues/new

    Briefly describe what you were doing, and include the last 20 lines from .coda-config/coda.log.
    And then paste the following:

    (monitor.ml.Error
  (Failure
    "Couldn't unregister mask with UUID 37efd9b4-6ec6-30b3-9e9a-85f849faa87e from parent f85ce994-cae9-3c6d-fb9e-c900be998eb5, mask not registered with that parent")
```

**Comment by ValarDragon** [comment]
Same error, it was immediately preceded by

**Comment by ValarDragon** [comment]
Got the same crash. It was preceded by: 
```
2019-07-24 22:21:55 UTC [Info] Connected to some peers [[host : 148.64.99.117, discovery_port : 7303, communication_port : 7302]]
```
The node was running a validator and a snark worker.

**Comment by boscro** [comment]
Got the same crash.  It was preceded by this:
`2019-07-25 01:59:25 UTC [Warn] Not rebroadcasting block $state_hash because it was received "2 slots too late"
        state_hash: "34483113080332616523439674397437260819089195443127422230748633569541025808399985544248693812097916884280378811401739427935596759081673847182121412614445134173660476632311222773220393991652783096788521912997663190999105939727605"
2019-07-25 01:59:26 UTC [Warn] Not rebroadcasting block $state_hash because it was received "2 slots too late"
        state_hash: "24194555502368735930993100492110271037860871302346606581645017095928766230259562998923158509965167692328012776249508690952271939424960222518843242504337716629891033241617084999505976127217759262786116547693594429600445926309905"


  ☠  Coda Daemon crashed. The Coda Protocol developers would like to know why!
`
The node was running a validator and a snark worker.

**Comment by AlexanderYudin** [comment]
☠  Coda Daemon crashed. The Coda Protocol developers would like to know why!

  Please:
    Open an issue:
      https://github.com/CodaProtocol/coda/issues/new

    Briefly describe what you were doing, and include the last 20 lines from .co                                                                                                                                                             da-config/coda.log.
    And then paste the following:

    (monitor.ml.Error
  (runtime-lib/runtime.ml.E
    "Merkle roots differ after transfer: got unexpected result"
    ((expected
       3893269504046759610686640273459744366906338144662816783892912922295865035                                                                                                                                                             94026600181694925421142285798570168096633628787029676911401148838806352978433375                                                                                                                                                             26424344049903879831696969707777320618038873222528382381697025453708056747)
      (got
        166148763610250116032306976950853705543012753235844797875908664113023920                                                                                                                                                             47573659534418116050177771442629407029327750673002858692494540135237949546362363                                                                                                                                                             166695549101792654595707467823915385172461099965870109734933066241071519040)
      (Loc lib/coda_base/ledger_transfer.ml:25:19)))
  ("Raised at file \"runtime-lib/runtime.ml\", line 28, characters 28-53"
    "Called from file \"runtime-lib/runtime.ml\" (inlined), line 105, characters                                                                                                                                                              27-78"
    "Called from file \"lib/coda_base/ledger_transfer.ml\", line 25, characters                                                                                                                                                              19-32"
    "Called from file \"lib/coda_lib/coda_lib.ml\", line 352, characters 4-73"
    "Called from file \"lib/coda_lib/coda_lib.ml\", line 394, characters 12-52"
    "Called from file \"src/deferred0.ml\", line 56, characters 64-69"
    "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5"
    "Called from file \"src/job_queue.ml\", line 170, characters 6-47"
    "Caught by monitor coda"))
root@232323:~#

**Comment by garethtdavies** [comment]
Similar crash with the error:

```("Merkle_mask.Masking_merkle_tree.Make(Inputs).Attached.Dangling_parent_reference(\"42ca5a71-f993-3b1f-3a88-7a504b15e7ec\")")
  ("Raised at file \"lib/merkle_mask/masking_merkle_tree.ml\", line 110, characters 10-50"
    "Called from file \"lib/merkle_mask/masking_merkle_tree.ml\", line 240, characters 6-26"
    "Called from file \"lib/coda_base/ledger.ml\", line 198, characters 25-38"
    "Called from file \"lib/staged_ledger/staged_ledger.ml\", line 443, characters 6-31"
    "Called from file \"lib/staged_ledger/staged_ledger.ml\", line 482, characters 6-108"
    "Called from file \"src/deferred0.ml\", line 56, characters 64-69"
    "Called from file \"src/job_queue.ml\" (inlined), line 131, characters 2-5"
    "Called from file \"src/job_queue.ml\", line 170, characters 6-47"
    "Caught by monitor coda"))
```

Log file:

```
{"timestamp":"2019-07-25 20:24:14.152192Z","level":"Trace","source":{"module":"Transition_frontier","location":"File \"lib/transition_frontier/transition_frontier.ml\", line 647, characters 56-63"},"message":"Common ancestor: $state_hash","metadata":{"pid":30396,"state_hash":"3097474792007227320451202494806326561763390055132341573789760830771914052549445932343856357033188392244779621015802844269718755660395038337768206508644393849108533288017756157070255369440014810587090730135909782813807554866778"}}
{"timestamp":"2019-07-25 20:24:14.157254Z","level":"Debug","source":{"module":"Transition_frontier","location":"File \"lib/transition_frontier/transition_frontier.ml\", line 737, characters 60-67"},"message":"added 1 breadcrumbs and removed 0 making path to new best tip","metadata":{"new_breadcrumbs":[{"transition_with_hash":{"data":{"protocol_state":{"previous_state_hash":"3097474792007227320451202494806326561763390055132341573789760830771914052549445932343856357033188392244779621015802844269718755660395038337768206508644393849108533288017756157070255369440014810587090730135909782813807554866778","body":{"blockchain_state":{"staged_ledger_hash":{"non_snark":{"ledger_hash":"992486135471451440221433866902150115401819151374446481698163691315585411317913687050410682002994900267433908596899706400678927202529862560527521131416157520157054636813228577415134548897578786797811972353539485462924337984267","aux_hash":"8K7CWEja2NiXweqWUdqw3VaPKLLj7b21KaZjw4XDXY1jGHk9NHc8i","pending_coinbase_aux":"5LPkRZ9mMyUCQHRBfJSeGWSpypnxep3Ty5ZRBHyCQcHT8YVHaNV"},"pending_coinbase_hash":"5014925768570304323313911546816900649331624619244898241660272467952011831937125097494351384126628067488559341022264957210021498230412477492606665226541939656859784462670352160180635748382228721952574178870958718258488445561401"},"snarked_ledger_hash":"15633465045921724886831983166170481348577672935883348699425681520384447962172644874381890632287461239811369613541058124803235807120002461997527831483283433466590211251708200481198340514679248630205447328357565152640165739466499","timestamp":"1564085702165"},"consensus_state":{"blockchain_length":"451","epoch_count":"6","min_epoch_length":"62","last_vrf_output":"6ycLHzaRGRRYTLkbvrDrULE9EBNALtfmT5i71eTU7oHwkQohcL","total_currency":10117240,"curr_epoch":"6","curr_slot":"69","staking_epoch_data":{"ledger":{"hash":"16614876361025011603230697695085370554301275323584479787590866411302392047573659534418116050177771442629407029327750673002858692494540135237949546362363166695549101792654595707467823915385172461099965870109734933066241071519040","total_currency":10114280},"seed":"36713587514898153795081577044342906299359651617051762021547489175678034613480024546086421245128390412210286606170183859615174751170303093099912269414594528164240669608486122055927429218369857960576767361879411367195116040465735","start_checkpoint":"26397546725798828302303363561457531545785216820231651548995049516711592732562755478989820181523748891871716831360575604174223468165368908073311698624444681057227841063960097562595868603272409646610641708943469655922611274611941","lock_checkpoint":"37222162346680766138848064039017266989516649585161096877415560975317815933550776162509362428188221319850504134047604915336104004176210961349078012277158976311703409913043167670202746614830353115489167560125790949422065818865375","epoch_length":"67"},"next_epoch_data":{"ledger":{"hash":"16614876361025011603230697695085370554301275323584479787590866411302392047573659534418116050177771442629407029327750673002858692494540135237949546362363166695549101792654595707467823915385172461099965870109734933066241071519040","total_currency":10114280},"seed":"11208977063950142350157768284178127890904349063477357324948739993956642229533400654921475932012546011712549756984162531531426187656258395607456412019587792875303345966217981690371965266985025240548881339407232853761379198964271","start_checkpoint":"39400081402700801099461373041848191661358937821974557536101841741030611341824624144050504330730326428257692352942377860767671268903324610900469867109867963855590321947919345915212472616661053107625632113021893444564454929692765","lock_checkpoint":"32864343320560526957140471911241448063508918357219589800053739393583646477913234762517754899515909235682887171638292944770034079448433265327130611261653080615417141171113225957047102409232804574472600789850447165739654749250410","epoch_length":"59"},"has_ancestor_in_same_checkpoint_window":true,"checkpoints":{"data":{"prefix":["36782869935965414222803766111589106059928653266006562368237491770920419223196351921290099024431045678696270371899077719754582744324683342090263286949885333652891185465016898405769609259698781267121722885844488370666255009246289"],"tail":"0"},"hash":"12062717558525538699047780862059193266924556459560851006160860343216576123787521205271672413556467062262639340546367611173236998289502441061812779352640732323492364704217327960088370937391626808596131208187008001457662385788416"}}}},"protocol_state_proof":"<opaque>","staged_ledger_diff":"<opaque>"},"hash":"32945676608331067382997143810546765368503946256678919936803839393107277277893196662258922390219185503924766131943814570759308439680676306863274266009778638069337986309654926193793553637115170777357748224930515493847370325580106"},"staged_ledger":"<opaque>","just_emitted_a_proof":false}],"old_breadcrumbs":[],"pid":30396}}
{"timestamp":"2019-07-25 20:24:14.157371Z","level":"Debug","source":{"module":"Transition_frontier","location":"File \"lib/transition_frontier/transition_frontier.ml\", line 753, characters 64-71"},"message":"Moving the root of the transition frontier. The new node is $distance_to_root blocks after the root, which exceeds the max length of $max_length.","metadata":{"distance_to_parent":4,"max_length":3,"pid":30396}}
{"timestamp":"2019-07-25 20:24:14.159274Z","level":"Trace","source":{"module":"Transition_frontier","location":"File \"lib/transition_frontier/transition_frontier.ml\", line 790, characters 64-71"},"message":"Removing $length_of_garbage nodes because the old root is one of their ancestors and we're deleting it","metadata":{"all_garbage":[],"garbage_direct_descendants":[],"length_of_garbage":0,"local_state":{"staking_epoch_snapshot":{"ledger_hash":"16614876361025011603230697695085370554301275323584479787590866411302392047573659534418116050177771442629407029327750673002858692494540135237949546362363166695549101792654595707467823915385172461099965870109734933066241071519040","delegators":{}},"next_epoch_snapshot":{"ledger_hash":"16614876361025011603230697695085370554301275323584479787590866411302392047573659534418116050177771442629407029327750673002858692494540135237949546362363166695549101792654595707467823915385172461099965870109734933066241071519040","delegators":{}},"last_checked_slot_and_epoch":{},"genesis_epoch_snapshot":{"ledger_hash":"38932695040467596106866402734597443669063381446628167838929129222958650359402660018169492542114228579857016809663362878702967691140114883880635297843337526424344049903879831696969707777320618038873222528382381697025453708056747","delegators":{}}},"pid":30396}}
{"timestamp":"2019-07-25 20:24:16.042805Z","level":"Warn","source":{"module":"Coda_lib","location":"File \"lib/coda_lib/coda_lib.ml\", line 526, characters 33-40"},"message":"Not rebroadcasting block $state_hash because it was received $timing","metadata":{"external_transition":{"protocol_state":{"previous_state_hash":"1327990808795495380622109199179630757819159518366641125725812593929687533289673119224672510447382200121257077731169814309692824922918643578604676341524578947618570513848529233944661377603399455633165278871937271599795824838662","body":{"blockchain_state":{"staged_ledger_hash":{"non_snark":{"ledger_hash":"9200028925640524667037909994882066855427258605443851239891714525884244136604508978736542972499850644759629347355097038561068840768378895910218329358415281733587314186564879072526902999587167815425470344374740390222082488492823","aux_hash":"8pfDMxjwFmknZjuNrxuLCiksQAf1aMgCDTX4DrxymgLGWV9bjnVZv","pending_coinbase_aux":"5LPkRZ9mMyUCQHRBfJSeGWSpypnxep3Ty5ZRBHyCQcHT8YVHaNV"},"pending_coinbase_hash":"34437111853083334119866775009810257654290099025831455191107805970890629604654414209748255140573144929132677955524396215874914277482049427234853273086161555367325332868279098838028529267939527128559561312775030555894279564771181"},"snarked_ledger_hash":"15633465045921724886831983166170481348577672935883348699425681520384447962172644874381890632287461239811369613541058124803235807120002461997527831483283433466590211251708200481198340514679248630205447328357565152640165739466499","timestamp":"1564085401976"},"consensus_state":{"blockchain_length":"450","epoch_count":"6","min_epoch_length":"62","last_vrf_output":"7XuTPeJ1c3aqr6pp4LdAkux7x345VaUuBf1ATLPqsRiushC1aG","total_currency":10117240,"curr_epoch":"6","curr_slot":"68","staking_epoch_data":{"ledger":{"hash":"16614876361025011603230697695085370554301275323584479787590866411302392047573659534418116050177771442629407029327750673002858692494540135237949546362363166695549101792654595707467823915385172461099965870109734933066241071519040","total_currency":10114280},"seed":"36713587514898153795081577044342906299359651617051762021547489175678034613480024546086421245128390412210286606170183859615174751170303093099912269414594528164240669608486122055927429218369857960576767361879411367195116040465735","start_checkpoint":"26397546725798828302303363561457531545785216820231651548995049516711592732562755478989820181523748891871716831360575604174223468165368908073311698624444681057227841063960097562595868603272409646610641708943469655922611274611941","lock_checkpoint":"37222162346680766138848064039017266989516649585161096877415560975317815933550776162509362428188221319850504134047604915336104004176210961349078012277158976311703409913043167670202746614830353115489167560125790949422065818865375","epoch_length":"67"},"next_epoch_data":{"ledger":{"hash":"16614876361025011603230697695085370554301275323584479787590866411302392047573659534418116050177771442629407029327750673002858692494540135237949546362363166695549101792654595707467823915385172461099965870109734933066241071519040","total_currency":10114280},"seed":"11208977063950142350157768284178127890904349063477357324948739993956642229533400654921475932012546011712549756984162531531426187656258395607456412019587792875303345966217981690371965266985025240548881339407232853761379198964271","start_checkpoint":"39400081402700801099461373041848191661358937821974557536101841741030611341824624144050504330730326428257692352942377860767671268903324610900469867109867963855590321947919345915212472616661053107625632113021893444564454929692765","lock_checkpoint":"32864343320560526957140471911241448063508918357219589800053739393583646477913234762517754899515909235682887171638292944770034079448433265327130611261653080615417141171113225957047102409232804574472600789850447165739654749250410","epoch_length":"58"},"has_ancestor_in_same_checkpoint_window":true,"checkpoints":{"data":{"prefix":["36782869935965414222803766111589106059928653266006562368237491770920419223196351921290099024431045678696270371899077719754582744324683342090263286949885333652891185465016898405769609259698781267121722885844488370666255009246289"],"tail":"0"},"hash":"12062717558525538699047780862059193266924556459560851006160860343216576123787521205271672413556467062262639340546367611173236998289502441061812779352640732323492364704217327960088370937391626808596131208187008001457662385788416"}}}},"protocol_state_proof":"<opaque>","staged_ledger_diff":"<opaque>"},"pid":30396,"state_hash":"3097474792007227320451202494806326561763390055132341573789760830771914052549445932343856357033188392244779621015802844269718755660395038337768206508644393849108533288017756157070255369440014810587090730135909782813807554866778","timing":"2 slots too late"}}
{"timestamp":"2019-07-25 20:24:16.082448Z","level":"Trace","source":{"module":"Network_pool__Transaction_pool","location":"File \"lib/network_pool/transaction_pool.ml\", line 161, characters 58-65"},"message":"Diff: removed: $removed added: $added from best tip","metadata":{"added":[{"payload":{"common":{"fee":1,"nonce":"22","memo":"2pmu64f2x97tNiDXMycnLwBSECDKbX77MTXVWVsG8hcRFsedhXDWWq"},"body":["Payment",{"receiver":"CfKCq2qU5ZDc8k7S4SjXojKCP22hoqBZBJY4oFKVUEzSMdRw2Aqvpq9jEqRumbtz4qNchbE1wWW5aHnsGwdub8Bc5p4Byh9FdJwss2hJG8TjfGfefEMMx7FsC8ZUh5Nyskneber3SPWM9","amount":0}]},"sender":"JzYE7smEuBxX9B9vznLLcngBq4uGM8CyWbuLZHSFWvahC4718mDW1gjqxeeJMxE6S9HaY4uJtE2cnSHDFzaYWQT9yfYPUN8ktvar3HYLhEpwiTsa7hQPgaszUBajveYd8wUMmQf6sA6gRbCxz7S7rtmmAskFcoV1PzxGifx1ageTYgQZAavqEWLrSUYjkd7MFYhJzZgm3XdxeuuifG5C76YvzTRWQt3iXdXwF29kvz7Gu356Le8syLTrDGqmTRhL4TbtgjEMLFJLNmaD","signature":"EXPeY8rAiAFixefLaKNM8Jok5F5VbtxXFNdu45LXq4Tt5FeFtaQd5hoDqaqmFGb5aPqWofiuV1gUfrZbHZegNLVkUZyDmLgvLUM9eUHEJ8L712nefL45cHC7rzgb5eGRAkp2gerk4G7aLjSMFaPQtnGZZGGMNBxqqK7WiijDgf3DmCNJwuux19apxHjLEEGqREKEAfxHfQtsQDnf21fW2QrAPetdHyy5PN7RCfJsSBWAbRihYzQ4xEb7AFSsR77oL9dJn3H8HSFD6VbL"}],"pid":30396,"removed":[]}}
{"timestamp":"2019-07-25 20:24:16.082808Z","level":"Debug","source":{"module":"Network_pool__Transaction_pool","location":"File \"lib/network_pool/transaction_pool.ml\", line 204, characters 58-65"},"message":"Finished handling diff. Old pool size 0, new pool size 0. Dropped 0 commands during backtracking to maintain max size.","metadata":{"pid":30396}}
{"timestamp":"2019-07-25 20:24:16.082948Z","level":"Debug","source":{"module":"Coda","location":"File \"app/cli/src/coda.ml\", line 467, characters 63-70"},"message":"Long async cycle: $span milliseconds","metadata":{"pid":30396,"span":"7.134387091s"}}
{"timestamp":"2019-07-25 20:24:16.083132Z","level":"Debug","source":{"module":"Coda_run","location":"File \"app/cli/src/coda_run.ml\", line 70, characters 52-59"},"message":"Successfully wrote the visualization of the registered masks at location: /home/gareth/.coda-config/registered_masks.dot.","metadata":{"coda_run":"Logging state before program ends","pid":30396}}
{"timestamp":"2019-07-25 20:24:16.189599Z","level":"Debug","source":{"module":"Coda_run","location":"File \"app/cli/src/coda_run.ml\", line 80, characters 58-65"},"message":"Successfully wrote the visualization of the transition frontier at location: /home/gareth/.coda-config/frontier.dot.","metadata":{"coda_run":"Logging state before program ends","pid":30396}}
```

**Comment by o1pranay** [comment]
Fixed this issue in #3009

**Comment by ghost-not-in-the-shell** [comment]
I saw another crash related to unregistered mask. https://187264-114581326-gh.circle-artifacts.com/0/home/opam/project/test_logs/1564702675/test_postake_holy_grail/coda-restarts-and-txns-holy-grail%20-num-proposers%205.log

I think it might be that we are unregistering mask in a wrong order.

**Comment by o1pranay** [comment]
@ghost-not-in-the-shell did you see this bug in a more recent commit than the one that supposedly fixed this issue?

**Comment by ghost-not-in-the-shell** [comment]
Yes.

**Comment by o1pranay** [comment]
should be resolved by #3087 - closing this issue out
